test_common/harness/os_helpers.cpp

```diff
@@ -208,7 +208,8 @@ static std::string _err_msg(int err, int level)
     */
 
 #if (defined(__ANDROID__) && __ANDROID_API__ < 23)                             \
-    || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE)
+    || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE)  \
+    || defined(_GNU_SOURCE) && !defined(__GLIBC__)
```

Comment:
Suggesting to add some parentheses for consistency with the preceding clauses:
```suggestion
    || (defined(_GNU_SOURCE) && !defined(__GLIBC__))
```